actions/changelog-generate/action.yaml

```diff
@@ -0,0 +1,128 @@
+name: "Generate changelog"
+description: "Generate changelog after release"
+inputs:
+  # required inputs
+  github-token:
+    description: "The GitHub token for committing the changes."
+    required: true
+  new-tag:
+    description: Version after bump
+    required: true
+
+  # unrequired inputs
+  changelog-file:
+    description: Path to the changelog file in the GitHub repository
+    required: false
+    default: "CHANGELOG.md"
+  commitMode:
+    description: "Special configuration for projects which work without PRs."
+    required: false
+    default: "false"
+  config:
+    description: Path to the changelog config JSON file
+    required: false
+    default: "./.github/changelog-config.json"
+  fetchReleaseInformation:
+    description: "Will enable fetching additional release information from tags."
+    required: false
+    default: "false"
+  fetchReviewers:
+    description: "Will enable fetching the users/reviewers who approved the PR"
+    required: false
+    default: "false"
+  old-tag:
+    description: Previous version. Let empty for releases
+    required: false
+    default: ""
+outputs:
+  merged-changelog:
+    description: "Changelog containing listing of all single changelogs"
+    value: ${{ steps.output-generator.outputs.merged-changelog }}
+  single-changelog:
+    description: "Changelog containing changes of the latest tag"
+    value: ${{ steps.output-generator.outputs.single-changelog }}
+runs:
+  using: "composite"
+  steps:
+    - name: "Set tags"
+      id: setTags
+      run: |
+        tag_list=$(git describe --tags) || tag_list=""
+
+        # to-tag
+        if [[ "$tag_list" == *"${{ inputs.new-tag }}"* ]]; then
+          echo  "totag=${{ inputs.new-tag }}">> $GITHUB_OUTPUT
+        else
+          echo  "totag=$(git rev-parse HEAD)">> $GITHUB_OUTPUT 
+        fi
+
+        #from-tag
+        setfrom=""
+
+        if [[ -n "${{ inputs.old-tag }}" ]] && [[ "$tag_list" == *"${{ inputs.old-tag }}"* ]]; then
+        # tag provided & provided tag exists
+            setfrom=${{ inputs.old-tag }}
+        else
+        # no tag provided or a non-existing tag was provided 
+          if [ -n "$tag_list" ]; then 
+            # if any tag exists then set from-tag to the latest tag
+            setfrom=$(git describe --tags `git rev-list --tags --max-count=1`)
+          else
+            # if no tag exists then use the frist commit as fromtag
+            setfrom=$(git rev-list --max-parents=0 HEAD)
+          fi
+        fi
+        echo  "fromtag=$setfrom">> $GITHUB_OUTPUT
+      shell: bash
+
+    - name: "Create changelog"
+      id: build_changelog
+      uses: mikepenz/release-changelog-builder-action@v3.7.0
+      with:
+        token: "${{ inputs.github-token }}"
+        configuration: "${{ inputs.config }}"
+        fromTag: "${{ steps.setTags.outputs.fromtag }}"
+        toTag: "${{ steps.setTags.outputs.totag }}"
+        outputFile: changes.md
```

Comment:
`changes.md` might exist in the repository itself, but you can use a tmp dir for this. Apparently GitHub provides a custom one that you can access via ` ${{ runner.temp }}`, see https://docs.github.com/en/actions/learn-github-actions/contexts#example-usage-of-the-runner-context